src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -951,6 +958,29 @@ internal void ResolveConflict([NotNull]string docID)
                 }
             });
         }
+
+        internal void SchedulePurgeExpired(TimeSpan delay)
+        {
+            var nextExpiration = Native.c4db_nextDocExpiration(_c4db);
+            if (nextExpiration > 0) {
+                var delta = (DateTimeOffset.FromUnixTimeMilliseconds((long)nextExpiration) - DateTimeOffset.UtcNow).Add(TimeSpan.FromSeconds(1));
```

Comment:
What is the reason for adding one second?

Reply:
because of SchedulePurgeExpired(TimeSpan.FromSeconds(1));

Reply:
needs 1 sec increment or it will stuck in a loop